### PR TITLE
Rn version switcher

### PR DESCRIPTION
**Pull request by jacoblogan**
#### Description of changes:
Add version switcher to RN and all web platforms

staging site: https://rn-version-switcher.dmhjrabi3pkql.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/styles/platform-navigator.scss [hunk 1]
```diff
@@ -6,6 +6,8 @@
   &__button {
     display: flex;
     justify-content: space-between;
+    white-space: nowrap;
```
Comment (jacoblogan) [inline src/styles/platform-navigator.scss:9]: React Native title was wrapping

File: src/components/PlatformNavigator/index.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export function PlatformNavigator({ currentPlatform, isPrev }) {
             flex="1 1 0"
             paddingRight="xs"
           >
-            <Flex as="span" alignItems="center" gap="small">
+            <Flex as="span" alignItems="center" gap="xs">
```
Comment (jacoblogan) [inline src/components/PlatformNavigator/index.tsx:64]: React Native looked very cramped trying to buy just a little more space

**Review by sowmyadev-kota — DISMISSED** [review]

File: src/styles/platform-navigator.scss (hunk 1 above)
Comment (timngyn) [reply src/styles/platform-navigator.scss:9]: I'm wondering if we should just decrease the font size or maybe make the arrow smaller? It still looks really cramped:

image

**Review by hbuchel — APPROVED** [review]
Thanks for the updates, lgtm!

**Review by timngyn — APPROVED** [review]

**Pull request merged**